### PR TITLE
test(Omnichannel): Fix department flaky test

### PR DESCRIPTION
**Pull request by MartinSchoeler**
[FLAKY-735](https://rocketchat.atlassian.net/browse/FLAKY-735)

## Further comments

[FLAKY-735]: https://rocketchat.atlassian.net/browse/FLAKY-735?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/meteor/tests/e2e/omnichannel/omnichannel-departaments.spec.ts [hunk 1]
```diff
@@ -22,101 +22,105 @@ test.describe('OC - Manage Departments', () => {
...
 		// turn on department removal
-		const statusCode = (await api.post('/settings/Omnichannel_enable_department_removal', { value: true })).status();
-		await expect(statusCode).toBe(200);
+		await api.post('/settings/Omnichannel_enable_department_removal', { value: true });
 	});
 
 	test.afterAll(async ({ api }) => {
 		// turn off department removal
-		const statusCode = (await api.post('/settings/Omnichannel_enable_department_removal', { value: false })).status();
-		await expect(statusCode).toBe(200);
```
Comment (dougfabris) [inline apps/meteor/tests/e2e/omnichannel/omnichannel-departaments.spec.ts:32]: why?

File: apps/meteor/tests/e2e/omnichannel/omnichannel-departaments.spec.ts [hunk 2]
```diff
@@ -22,101 +22,105 @@ test.describe('OC - Manage Departments', () => {
...
-		await test.step('expect to delete department', async () => {
-			await poOmnichannelDepartments.search(departmentName);
-			await poOmnichannelDepartments.selectedDepartmentMenu(departmentName).click();
-			await poOmnichannelDepartments.menuDeleteOption.click();
+			await test.step('expect create new department', async () => {
+				await poOmnichannelDepartments.btnEnabled.click();
+				await poOmnichannelDepartments.inputName.fill(departmentName);
+				await poOmnichannelDepartments.inputEmail.fill(faker.internet.email());
+				await poOmnichannelDepartments.btnSave.click();
+				await poOmnichannelDepartments.btnCloseToastSuccess.click();
```
Comment (dougfabris) [inline apps/meteor/tests/e2e/omnichannel/omnichannel-departaments.spec.ts:73]: Indeed, no need to close this `ToastBar`

File: apps/meteor/tests/e2e/omnichannel/omnichannel-departaments.spec.ts [hunk 3]
```diff
@@ -22,101 +22,105 @@ test.describe('OC - Manage Departments', () => {
...
-					await expect(poOmnichannelDepartments.btnModalConfirmDelete).toBeDisabled();
-					await poOmnichannelDepartments.inputModalConfirmDelete.fill('someramdomname');
-					await expect(poOmnichannelDepartments.btnModalConfirmDelete).toBeDisabled();
+			await test.step('expect to delete department', async () => {
+				await poOmnichannelDepartments.search(departmentName);
+				await poOmnichannelDepartments.selectedDepartmentMenu(departmentName).click();
+				await poOmnichannelDepartments.menuDeleteOption.click();
+
+				await test.step('expect confirm delete department', async () => {
+					await expect(poOmnichannelDepartments.modalConfirmDelete).toBeVisible();
```
Comment (dougfabris) [inline apps/meteor/tests/e2e/omnichannel/omnichannel-departaments.spec.ts:85]:
```suggestion

```

No need to do this assertion

**Review by dougfabris — CHANGES_REQUESTED** [review]

**Review by ggazzo — APPROVED** [review]

File: apps/meteor/tests/e2e/omnichannel/omnichannel-departaments.spec.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (dougfabris) [reply apps/meteor/tests/e2e/omnichannel/omnichannel-departaments.spec.ts]: fixed!

**Pull request merged**